### PR TITLE
feat(ffi): Add a notification_mode property to room_info

### PR DESCRIPTION
**Pull request by nimau**
This PR adds the notification mode in `room_info` required by https://github.com/vector-im/element-x-ios/issues/1035

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/room_info.rs [hunk 1]
```diff
@@ -57,6 +64,33 @@ impl RoomInfo {
...
+
+/// Get the notification mode for a given `Room`
+async fn notification_mode(room: &matrix_sdk::Room) -> Option<RoomNotificationMode> {
+    if !matches!(room.state(), RoomState::Joined) {
+        return None;
+    }
+    let notification_settings = room.client().notification_settings().await;
+
+    // Get the user-defined mode if available
+    let sdk_mode =
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room_info.rs:80]: It's incorrect to have logic like this in the FFI layer, this should go elsewhere in the SDK base crate.

**Review by bnjbvr — COMMENTED** [review]
Makes sense; this isn't the right place to put any kind of logic, though, so it should be put elsewhere.

File: bindings/matrix-sdk-ffi/src/room_info.rs (hunk 1 above)
Comment (nimau) [reply bindings/matrix-sdk-ffi/src/room_info.rs]: @bnjbvr, I've moved the logic into the SDK, let me know if that works for you.

**Review by bnjbvr — APPROVED** [review]
Nice!

File: crates/matrix-sdk/src/room/mod.rs [hunk 2]
```diff
@@ -2359,6 +2360,32 @@ impl Room {
...
+            return None;
+        }
+        let notification_settings = self.client().notification_settings().await;
+
+        // Get the user-defined mode if available
+        let notification_mode =
+            notification_settings.get_user_defined_room_notification_mode(self.room_id()).await;
+        if notification_mode.is_some() {
+            return notification_mode;
+        } else if let Ok(is_encrypted) = self.is_encrypted().await {
```
[the remark this answers is not in the document]
Comment (nimau) [reply crates/matrix-sdk/src/room/mod.rs]: 👍

**Review by bnjbvr — APPROVED** [review]

**Pull request merged**